crates/tuono/src/source_builder.rs

```diff
@@ -106,42 +114,57 @@ impl SourceBuilder {
         Ok(())
     }
 
-    fn generate_axum_source(&self) -> String {
+    fn generate_axum_source(&self) -> (Option<String>, String) {
```

Comment:
I'd a top level comment to describe what is the returned tuple

```suggestion
    /// Returns a tuple containing:
    /// - The optional `.tuono/main.rs` file content
    /// - The dynamically generated `.tuono/router.rs` file content
    fn generate_axum_source(&self) -> (Option<String>, String) {
```